cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -239,6 +240,22 @@ class LeadImpl extends ServerImpl with Lead
     }
   }
 
+  private def checkAuthProvider(props: Properties): Unit = {
+    var authP = props.getProperty(Attribute.AUTH_PROVIDER)
+    if (authP == null) {
+      authP = props.getProperty(Attribute.SERVER_AUTH_PROVIDER)
+    }
```

Comment:
I now remember that SERVER_AUTH_PROVIDER check was added because if user specifies auth-provider as java system property (-J-D-auth-provider), it sets the SERVER_AUTH_PROVIDER property.